### ISSUE TITLE
Speech to text issue

### ISSUE THREAD

**Issue by gbissio** [Azure-Samples/chat-with-your-data-solution-accelerator#1289]
# Describe the bug
Every time that I click on the microphone within the chat the screen is showing "Please Wait" all the time with no action.

# Expected behavior

I'm testing the speech behavior because my customer needs to use it but I couldn't.

# Debugging information

## Steps to reproduce
1. Go to the site
2. Click on micrphone
3. a "Please Wait" message appear
4. Nothing happen.

## Screenshots

Besides... I enabled the option to log the messages and conversations but I don't know how can I get the information... How can I query that?

**Comment by Govardhana-Microsoft** [comment]
@gbissio We are able to perform speech to text. could you please confirm whether you are able to allow 'use your microphone' popup. 
So, team will proceed with further investigation on this.

<img width="843" alt="Allow popup" src="https://github.com/user-attachments/assets/5fbef3f2-3b9e-491a-8e8f-630784cddc6e">

**Comment by gbissio** [comment]
Hey! 
Thank you for your answer! Yeap! I already did that:

![image](https://github.com/user-attachments/assets/415d72ca-9b7a-4080-b38b-dbe479003694)

I tried it with chrome, edge and with my phone with no luck. Can i check any particular log?

**Comment by gbissio** [comment]
Just in case I found this on the logs:

2024-09-09T12:52:38.203099190Z ERROR:create_app:Exception in /api/speech | HTTPSConnectionPool(host='east%20us%202.api.cognitive.microsoft.com', port=443): Max retries exceeded with url: /sts/v1.0/issueToken (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x79c89758ef50>: Failed to resolve 'east%20us%202.api.cognitive.microsoft.com' ([Errno -2] Name or service not known)"))

For some reason the speech is not using the real name

**Comment by Roopan-Microsoft** [comment]
Hey @gbissio, I feel you might have given the location as: **_east us_** while deploying, because I see the URL has some space "**east%20us**%202.api.cognitive.microsoft.com". Please check and reconfirm, we will also investigate it parallelly.
<img width="535" alt="image" src="https://github.com/user-attachments/assets/4005b277-33c8-4b3b-be77-e86b025a4079">

**Comment by gbissio** [comment]
![image](https://github.com/user-attachments/assets/49767016-fd0b-4d38-9949-06fb60e0bfdc)

Should I use Eastus2 everything togeher?

**Comment by Roopan-Microsoft** [comment]
yes, you can give us **eastus2**. Let me know if still not working. 

Happy to help :)

**Comment by Prasanjeet-Microsoft** [comment]
@gbissio Could you please let us know if you are still experiencing the issue? If not, could you kindly close this issue?
